Xamarin.PropertyEditing.Mac/Controls/BindingEditor/BindingEditorWindow.cs

```diff
@@ -0,0 +1,276 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using Xamarin.PropertyEditing.Common;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class BindingEditorWindow : BasePanel
+	{
+		private readonly CreateBindingViewModel viewModel;
+		private readonly PropertyEditorSelector editorSelector = new PropertyEditorSelector ();
+
+		private const string BindingPropertiesIdentifier = "BindingProperties";
+		private const string FlagPropertiesIdentifier = "FlagProperties";
+
+		internal CreateBindingViewModel ViewModel => this.viewModel;
+
+		internal BindingEditorWindow (IHostResourceProvider hostResources, PropertyViewModel propertyViewModel) 
+		{
+			this.viewModel = new CreateBindingViewModel (propertyViewModel.TargetPlatform, propertyViewModel.Editors.Single (), propertyViewModel.Property);
+
+			Title = this.viewModel.PropertyDisplay;
+			this.ButtonDone.Title = Properties.Resources.CreatBindingTitle;
+
+			foreach (BindingSource item in this.viewModel.BindingSources.Value) {
```

Comment:
This will throw an `NRE` if `BindingSources` hasn't loaded yet, it's an `AsyncValue<T>` which means `Value` will contain a default value until it's loaded. You should await the task on it before filling out this list.

Reply:
Sure.